# Code generated by lark_sdk_gen. DO NOT EDIT.

from tests.test_conf import app_all_permission
import unittest
import pylark
import pytest
from tests.test_helper import mock_get_tenant_access_token_failed


class TestACSSampleRequestFailed(unittest.TestCase):
    def __init__(self, *args, **kwargs):
        super(TestACSSampleRequestFailed, self).__init__(*args, **kwargs)
        self.cli = app_all_permission.ins()
        self.cli.auth.get_tenant_access_token = mock_get_tenant_access_token_failed
        self.cli.auth.get_app_access_token = mock_get_tenant_access_token_failed
        self.module_cli = self.cli.acs

    def test_request_failed_get_acs_access_record_photo(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_access_record_photo(
                pylark.GetACSAccessRecordPhotoReq()
            )

        assert "msg=failed" in f"{e}"

    def test_request_failed_get_acs_access_record_list(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_access_record_list(
                pylark.GetACSAccessRecordListReq()
            )

        assert "msg=failed" in f"{e}"

    def test_request_failed_get_acs_device_list(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_device_list(pylark.GetACSDeviceListReq())

        assert "msg=failed" in f"{e}"

    def test_request_failed_get_acs_user_face(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_user_face(pylark.GetACSUserFaceReq())

        assert "msg=failed" in f"{e}"

    def test_request_failed_update_acs_user_face(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.update_acs_user_face(pylark.UpdateACSUserFaceReq())

        assert "msg=failed" in f"{e}"

    def test_request_failed_get_acs_user(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_user(pylark.GetACSUserReq())

        assert "msg=failed" in f"{e}"

    def test_request_failed_update_acs_user(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.update_acs_user(pylark.UpdateACSUserReq())

        assert "msg=failed" in f"{e}"

    def test_request_failed_get_acs_user_list(self):
        with pytest.raises(pylark.PyLarkError) as e:
            self.module_cli.get_acs_user_list(pylark.GetACSUserListReq())

        assert "msg=failed" in f"{e}"
